packages/core/admin/admin/src/content-manager/pages/EditView/index.js

```diff
@@ -131,99 +136,101 @@ const EditView = ({
               <ContentLayout>
                 <Grid gap={4}>
                   <GridItem col={9} s={12}>
-                    <Stack spacing={6}>
-                      {formattedContentTypeLayout.map((row, index) => {
-                        if (isDynamicZone(row)) {
-                          const {
-                            0: {
-                              0: { name, fieldSchema, metadatas, labelAction },
-                            },
-                          } = row;
+                    <Suspense fallback={<LoadingIndicatorPage />}>
```

Comment:
Why do you need `<Suspense />` ?

Reply:
Hi soup! It's because the custom fields inputs are loaded via useLazy, which requires a suspense parent. And we thought it would be better to put it at the form level to have only a single loader even if there are several custom field inputs

Reply:
I thought it was for this indeed, can you try to avoir React.useMemo(() => React.lazy)) and do dynamic import in a dedicated component instead?

Reply:
As I mentioned in your previous comment, I'd be interested to know what are the risks of memo and lazy together

Because otherwise I think this current solution is interesting:

- it handles the loading state in common for all custom fields, to avoid having several spinners at once on the page. I don't think it would be possible if we handle the loading at the custom level
- it will let us easily lazy load other non-custom fields later, which is good for performance. For example, the big wysiwyg component could only be loaded if there's a rich text attribute

Reply:
Thanks @remidej for the explanations!

>  I'd be interested to know what are the risks of memo and lazy together

Honestly for this one, I am not able to provide any but the React documentation have never advertise to use it this way. I know that this is working but to me it feels like a big hack and I would dig into some other architectural patterns in order to load the custom fields into a separate bundle. You can do this by implementing a `LazyInputComponent` and use dynamic `import()` just like we do [here](https://github.com/strapi/strapi/blob/main/packages/core/admin/admin/src/utils/createRoute.js)


> Because otherwise I think this current solution is interesting:

it handles the loading state in common for all custom fields, to avoid having several spinners at once on the page. I don't think it would be possible if we handle the loading at the custom field level
it will let us easily lazy load other non-custom fields later, which is good for performance. For example, the big wysiwyg component could only be loaded if there's a rich text attribute

I agree with all the points this is why we have started to introduce a Field API in order to tackle all these cases, but I strongly that the Inputs component is the right place to do it.

With this implementation you are reloading the entire page when you have a custom field inside a component of a dynamic zone, you can try it out by doing the following:
- create a collection type and a a simple text field
- Create a dz and add a component that has the color picker make sure to add  the custom field only in the DZ and not at the CT level
- run the yarn build command in the getstarted app
- Go to the cm and create an entry
- click on the DZ and add your component, you'll see the reloading.

Reply:
I haven't found an example of lazy + memo used together either, but I don't think that means it would be a problem. On the react [rules of hook](https://reactjs.org/docs/hooks-rules.html#gatsby-focus-wrapper) page, they say it's fine to call any hook from a custom hook. So here I'd argue useMemo is a custom hook, just one that react provides. And I think using our own solution for something that in terms of functionality very similar to suspense+lazy doesn't sound great...

Since the release it tomorrow I don't think it would have been possible to make the change in time anyway. And we haven't gotten bug reports about this part so far so I it's safer to keep it for tomorrow.

The issue you bring up though is quite important, and something we want to fix now. But I found another fix which is to add a new suspense boundary at the dynamic zone level. The related PR is #14474

Reply:
I understand, but please make sure to investigate and provide a fix because the implementation is going to be problematic in the long term. I strongly believe that it is not the responsibility of the Inputs component to lazy load injected fields but more the responsibility of the library API. 
You can use the the Library to lazy load all the custom inputs when the app is mounting just like we do with the translations and since the bundles will always be injected React.lazy won't try to import the custom fields when inside the CM (or any other plugins).